Upload excel files to drive task

nemerna opened the pull request:
- Fix a Bug of Prepare-source-code taks deleting the ignore.err downloaded by Prepare-false-positives task
- Add new Task for uploading the Excel file to Gdrive based on Parameter/Configmap
Test Results:
https://console-openshift-console.apps.appeng.clusters.se-apps.redhat.com/k8s/ns/sast-ai/tekton.dev~v1~PipelineRun/sast-ai-workflow-pipeline-jesv4v/logs?taskName=upload-results-to-gdrive

Uploaded-Drive:
https://drive.google.com/drive/folders/1tRQT3I637FUR2KQcuACvtxxE3xslyRS5

Yael-F commented on deploy/tekton/tasks/excel_file_upload.yaml:
```diff
@@ -0,0 +1,322 @@
+        
+        # Execute the upload
+        echo "Executing Google Drive upload..."
+        source /tmp/venv/bin/activate
```
duplicate?

Yael-F commented on deploy/tekton/tasks/excel_file_upload.yaml:
```diff
@@ -0,0 +1,322 @@
+        
+        # Extract service account credentials from workspace
+        echo "Setting up service account credentials..."
+        cp "$(workspaces.google-sa-json-ws.path)/service_account.json" /tmp/service-account.json
```
$GOOGLE_APPLICATION_CREDENTIALS

Yael-F reviewed (commented):
This change looks really good and well-written! I enjoyed reviewing it. The logic is clear and the new functionality seems robust :rocket:

left some suggestions

JudeNiroshan commented on deploy/tekton/pipeline.yaml:
```diff
@@ -35,6 +35,10 @@ spec:
+    - name: GDRIVE_FOLDER_ID
+      type: string
+      description: "Google Drive folder ID for uploading SAST results (optional)"
+      default: "1-alhALixl2Fn2wIWUD_MRwfHhYgIZrQE"
```
Is this safe to be visible on a public repo?

nemerna commented:
@JudeNiroshan @Yael-F 
i have fixed the minor issues requested by @Yael-F 

also:
1) separate the script (found the configmap as the best overall option)
2) fix minor issues
3) removed the hard coded ServiceAccount and replaced with parameters
4) Added Retry logic as @JudeNiroshan requested

[test_results](https://console-openshift-console.apps.appeng.clusters.se-apps.redhat.com/k8s/ns/sast-ai/tekton.dev~v1~PipelineRun/sast-ai-workflow-pipeline-2kd9w1/logs?taskName=upload-results-to-gdrive)

nemerna commented on deploy/tekton/tasks/excel_file_upload.yaml:
```diff
@@ -0,0 +1,322 @@
+                media_body=media,
+                fields='id,name,webViewLink',
+                supportsAllDrives=True
+            ).execute()
```
added retry logic

nemerna commented on deploy/tekton/tasks/excel_file_upload.yaml:
```diff
@@ -0,0 +1,322 @@
+        echo "Target folder ID: $GDRIVE_FOLDER_ID"
+        
+        # Create Python script for Google Drive operations with Shared Drive support
+        cat > /tmp/gdrive_upload.py << 'EOF'
```
moved it to be a configmap instead for easier maintainability, simply done that cuz we may want to change things, we dont want a new PR everytime we want to work on modifying the Tekton tasks

JudeNiroshan reviewed (approved):
LGTM 🚀
